tests/e2e/e2e_test.go

```diff
@@ -193,93 +191,465 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
+		feeGrowthInsideAddress1Position1Last,             // feeGrowthInside from first and second swaps
+		feeGrowthGlobal,
+	)
+
+	// Assert
+	s.Require().Equal(
+		addr1BalancesBefore.AmountOf("uion").Add(feesUncollectedAddress1Position1_Swap3.TruncateInt()),
+		addr1BalancesAfter.AmountOf("uion"),
+	)
+
+	// Assert position that was active thoughout the whole swap:
+
+	// Track balance of address3
+	addr3BalancesBefore = s.addrBalance(node, address3)
+	node.CollectFees(address3, fmt.Sprintf("[%d]", minTick), "1400", poolID)
+	addr3BalancesAfter = s.addrBalance(node, address3)
+
+	// Assert that the balance changed only for tokenIn (uion)
+	s.assertBalancesInvariants(addr3BalancesBefore, addr3BalancesAfter, true, false)
+
+	// Was active throughout the whole swap, collects fees from 2 steps
+
+	// Step 1
+	feesUncollectedAddress3Position2_Swap3_Step1 := calculateUncollectedFees(
+		positionsAddress3[1].Position.Liquidity,
+		sdk.ZeroDec(), // no growth below
+		sdk.ZeroDec(), // no growth above
+		calculateFeeGrowthInside(feeGrowthGlobal_Swap2, sdk.ZeroDec(), sdk.ZeroDec()), // snapshot of fee growth at swap 2
+		feeGrowthGlobal.Sub(feeCharge_Swap3_Step2),                                    // step 1 hasn't earned fees from step 2
+	)
+
+	// Step 2
+	feesUncollectedAddress3Position2_Swap3_Step2 := calculateUncollectedFees(
+		positionsAddress3[1].Position.Liquidity,
+		sdk.ZeroDec(), // no growth below
+		sdk.ZeroDec(), // no growth above
+		calculateFeeGrowthInside(feeGrowthGlobal_Swap2, sdk.ZeroDec(), sdk.ZeroDec()), // snapshot of fee growth at swap 2
+		feeGrowthGlobal.Sub(feeCharge_Swap3_Step1),                                    // step 2 hasn't earned fees from step 1
+	)
+
+	// Calculate total fees acquired by address3 position2 from all swap steps
+	totalUncollectedFeesAddress3Position2 = feesUncollectedAddress3Position2_Swap3_Step1.Add(feesUncollectedAddress3Position2_Swap3_Step2)
+
+	// Assert
+	s.Require().Equal(
+		addr3BalancesBefore.AmountOf("uion").Add(totalUncollectedFeesAddress3Position2.TruncateInt()),
+		addr3BalancesAfter.AmountOf("uion"),
+	)
+
+	// Collect Fees: Sanity Checks
+
+	// Assert that positions, which were not included in swaps, were not affected
+
+	// Address3 Position2: [-1600; -200]
+	addr3BalancesBefore = s.addrBalance(node, address3)
+	node.CollectFees(address3, "[-1600]", "[-200]", poolID)
+	addr3BalancesAfter = s.addrBalance(node, address3)
+
+	// Assert that balances did not change for any token
+	s.assertBalancesInvariants(addr3BalancesBefore, addr3BalancesAfter, true, true)
+
+	// Address2's only position: [2200; 3420]
+	addr2BalancesBefore := s.addrBalance(node, address2)
+	node.CollectFees(address2, "2200", fmt.Sprintf("%d", maxTick), poolID)
+	addr2BalancesAfter := s.addrBalance(node, address2)
+
+	// Assert the balances did not change for every token
+	s.assertBalancesInvariants(addr2BalancesBefore, addr2BalancesAfter, true, true)
+
```

Comment:
Just curious: Do you think it makes sense to add tests that try to collect fees when there are none or trying to collect fees from different ranges then provided positions?

Reply:
actually the first one: 
> collect fees when there are none

was [already there](https://github.com/osmosis-labs/osmosis/pull/4119/files#diff-7d1fc1e232b55677582b51e2623aae6d8c15b5f0960ac6100fc7d733cb795dacR627) in the `Collect Fees: Sanity Checks` section

Reply:
hmmm @mattverse, we could add a test for asserting that trying to collect fees for position which is not yours is impossible, but that would horribly slow the test. 

the reason is that asserting that command has successfully passed is wrapped in `require.Eventually` and is capped by one minute. If we collect fees for a position that is not owned by a sender, we will currently have to wait for the whole minute to assert the failness

Other approach might be to inspect the logs of a container and check wether some logs were emitted that indicate failure of a command. I actually like this idea, because right now if a developer wants to run an e2e test, and something is failing inside, a person has to wait 1 minute more for the command to be declared failed, however, it might have failed, for example, after one second and just waited for 59 seconds more because `require.Eventually` said so. 

Let me know what you think on the other approach, it might improve dev-UX, but this is out of scope of this PR